File: 1768.cpp
#include <string>
using namespace std;
class Solution {
public:
  string mergeAlternately(string word1, string word2) {
    string res;
    for (int i = 0, j = 0; (i < word1.size() || j < word2.size());) {
      if (i < word1.size()) {
        res += word1[i++];
      }
      if (j < word2.size()) {
        res += word2[j++];
      }
    }
    return res;
  }
};

File: 1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        spliced = ""
        for c1, c2 in zip(word1, word2):
            spliced += c1 + c2
        if len(word1) < len(word2):
            spliced += word2[len(word1) :]
        elif len(word2) < len(word1):
            spliced += word1[len(word2) :]
        return spliced
